05/app.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react'
+import ReactDOM from 'react-dom'
+
+export default class App extends Component {
+    state = {
+        weatherData: null
+    }
+    constructor(props) {
+        super(props)
+    }
+    componentDidMount() {
+        this._getWeather();
+    }
+    _getWeather() {
+        const { lat, lon } = this.props;
+        const apiKey = 'c12166f6b699496195f7b2ca51407ce4';
+        const url = `https://api.weatherbit.io/v2.0/current?key=${apiKey}&lat=${lat}&lon=${lon}&lang=pl`
+
+        const weather = fetch(url)
+            .then(resp => (resp.ok) ? resp.json() : Promise.reject(resp))
+            .then(resp => {
+                this.setState({ weatherData: resp.data[0] })
+            })
+            .catch(err => console.log(err));
```

Comment:
To można by było wrzucić do osobnego pliku i tylko obsługiwać obietnicę w tym miejscu ;)

Reply:
osobny plik? tzn. jakiś fetch.js czy coś na ten kształt? ten nowy plik ma być komponentem? czy tylko funkcją nie wiem za bardzo jak to zrobić.